Can't build  keycloak 26.4.4 with quarkus.launch.rebuild=true

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

Can't build with quarkus.launch.rebuild=true

### Version

26.4.4

### Regression

- [ ] The issue is a regression

### Expected behavior

quarkus.launch.rebuild=true should finish without errors

### Actual behavior

bin/kc.sh -Dquarkus.launch.rebuild=true build

```
2025-11-07 17:56:09,111 TRACE [org.jboss.threads] (build-5) Thread "Thread[#24,build-5,5,]" exiting

2025-11-07 17:56:09,111 TRACE [org.jboss.threads] (build-58) Thread "Thread[#77,build-58,5,]" exiting

2025-11-07 17:56:09,112 TRACE [org.jboss.threads] (build-107) Thread "Thread[#126,build-107,5,]" exiting

2025-11-07 17:56:09,112 TRACE [org.jboss.threads] (build-27) Thread "Thread[#46,build-27,5,]" exiting

2025-11-07 17:56:09,119 DEBUG [io.quarkus.bootstrap.classloading.QuarkusClassLoader.lifecycle] (main) Closing class loader QuarkusClassLoader:Deployment Class Loader: PROD for keycloak@50ad3bc1: java.lang.RuntimeException: Created to log a stacktrace
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.close(QuarkusClassLoader.java:748)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createProductionApplication(AugmentActionImpl.java:180)
	at io.quarkus.deployment.mutability.ReaugmentTask.main(ReaugmentTask.java:70)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doReaugment(QuarkusEntryPoint.java:97)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:55)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:37)


2025-11-07 17:56:09,119 DEBUG [io.quarkus.bootstrap.classloading.QuarkusClassLoader.lifecycle] (main) Closing class loader QuarkusClassLoader:Augmentation Class Loader: PROD for keycloak@12d4bf7e: java.lang.RuntimeException: Created to log a stacktrace
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.close(QuarkusClassLoader.java:748)
	at io.quarkus.bootstrap.app.CuratedApplication.close(CuratedApplication.java:457)
	at io.quarkus.deployment.mutability.ReaugmentTask.main(ReaugmentTask.java:59)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doReaugment(QuarkusEntryPoint.java:97)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:55)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:37)


2025-11-07 17:56:09,121 TRACE [org.jboss.threads] (build-35) Thread "Thread[#54,build-35,5,]" exiting

Exception in thread "main" java.lang.reflect.UndeclaredThrowableException
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:43)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doReaugment(QuarkusEntryPoint.java:97)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:55)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:37)
Caused by: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#persistBuildTimeProperties threw an exception: java.lang.NullPointerException
	at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at java.base/java.util.Properties.put(Properties.java:1346)
	at org.keycloak.quarkus.runtime.cli.Picocli.getNonPersistedBuildTimeOptions(Picocli.java:582)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.persistBuildTimeProperties(KeycloakProcessor.java:792)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:874)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1586)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:376)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createProductionApplication(AugmentActionImpl.java:181)
	at io.quarkus.deployment.mutability.ReaugmentTask.main(ReaugmentTask.java:70)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	... 4 more
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#persistBuildTimeProperties threw an exception: java.lang.NullPointerException
	at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at java.base/java.util.Properties.put(Properties.java:1346)
	at org.keycloak.quarkus.runtime.cli.Picocli.getNonPersistedBuildTimeOptions(Picocli.java:582)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.persistBuildTimeProperties(KeycloakProcessor.java:792)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:874)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1586)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:122)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:78)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:372)
	... 7 more
Caused by: java.lang.NullPointerException
	at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at java.base/java.util.Properties.put(Properties.java:1346)
	at org.keycloak.quarkus.runtime.cli.Picocli.getNonPersistedBuildTimeOptions(Picocli.java:582)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.persistBuildTimeProperties(KeycloakProcessor.java:792)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:874)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1586)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

### How to Reproduce?

download latest version
run bin/kc.sh -Dquarkus.launch.rebuild=true build

### Anything else?

_No response_

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
@@ -565,6 +565,7 @@ public class Picocli {
             if (value.getValue() == null || value.getConfigSourceName() == null
                     || (quarkus && !value.getConfigSourceName().contains(QuarkusPropertiesConfigSource.NAME))) {
                 // only persist build options resolved from config sources and not default values
+                // instead we'll persist the profile (if set) because that may influence the defaults
                 return;
             }
             // since we're persisting all quarkus values, this may leak some runtime information - we don't want
@@ -589,8 +590,10 @@ public class Picocli {
         }
 
         String profile = org.keycloak.common.util.Environment.getProfile();
-        properties.put(org.keycloak.common.util.Environment.PROFILE, profile);
-        properties.put(LaunchMode.current().getProfileKey(), profile);
+        if (profile != null) {
+            properties.put(org.keycloak.common.util.Environment.PROFILE, profile);
+            properties.put(LaunchMode.current().getProfileKey(), profile);
+        }
 
         return properties;
     }

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java
@@ -311,7 +311,7 @@ public class PropertyMapper<T> {
             return configValue;
         }
 
-        // by unsetting the name this will not be seen as directly modified by the user
+        // by unsetting the configsource name this will not be seen as directly modified by the user
         return configValue.from().withName(name).withValue(mappedValue).withRawValue(value).withConfigSourceName(null).build();
     }
 

